lib/modules/device/DeviceService.ts

```diff
@@ -42,7 +42,13 @@ export class DeviceService {
   }
 
   private get impersonatedSdk() {
-    return (user: User) => this.sdk.as(user, { checkRights: false });
+    return (user: User) => {
+      if (user?._id) {
+        return this.sdk.as(user, { checkRights: false });
+      }
+
+      return this.sdk;
+    };
   }
```

Comment:
Since this is a very useful method should we provide it as well in the Framework?

Reply:
Well actually Kuzzle could just return the normal SDK if the `user` is null but I'm not sure about potential side effects for users who relied on the fact that the method will throw if an incorrect user is passed.

Also, I don't know where we could put this method in the framework. Let's see if we need it another time and talk about this